contract_comm/currency/currency.go

```diff
@@ -216,128 +246,100 @@ func (cc *CurrencyComparator) Cmp(val1 *big.Int, currency1 *common.Address, val2
 		return val1.Cmp(val2)
 	}
 
-	// Below code block is basically evaluating this comparison:
-	// val1 * exchangeRate1.Denominator/exchangeRate1.Numerator < val2 * exchangeRate2.Denominator/exchangeRate2.Numerator
-	// It will transform that comparison to this, to remove having to deal with fractional values.
-	// val1 * exchangeRate1.Denominator * exchangeRate2.Numerator < val2 * exchangeRate2.Denominator * exchangeRate1.Numerator
-	leftSide := new(big.Int).Mul(val1, new(big.Int).Mul(exchangeRate1.Denominator, exchangeRate2.Numerator))
-	rightSide := new(big.Int).Mul(val2, new(big.Int).Mul(exchangeRate2.Denominator, exchangeRate1.Numerator))
-	return leftSide.Cmp(rightSide)
+	return exchangeRate1.CmpValues(val1, val2, exchangeRate2)
 }
 
-func Cmp(val1 *big.Int, currency1 *common.Address, val2 *big.Int, currency2 *common.Address) int {
-	// Short circuit if the fee currency is the same. nil currency => native currency
-	if (currency1 == nil && currency2 == nil) || (currency1 != nil && currency2 != nil && *currency1 == *currency2) {
-		return val1.Cmp(val2)
+func (cc *CurrencyManager) ToGold(amount *big.Int, currency *common.Address) (*big.Int, error) {
```

Comment:
s/Gold/Celo. (Or s/Gold/NativeToken)
```suggestion
func (cc *CurrencyManager) ToCelo(amount *big.Int, currency *common.Address) (*big.Int, error) {
```